initca/initca_test.go

```diff
@@ -46,19 +47,23 @@ var csrFiles = []string{
 	"testdata/ecdsa256.csr",
 	"testdata/ecdsa384.csr",
 	"testdata/ecdsa521.csr",
+	"testdata/ed25519.csr",
 }
 
 var testRSACAFile = "testdata/5min-rsa.pem"
 var testRSACAKeyFile = "testdata/5min-rsa-key.pem"
 var testECDSACAFile = "testdata/5min-ecdsa.pem"
 var testECDSACAKeyFile = "testdata/5min-ecdsa-key.pem"
+var testED25519CAFile = "testdata/5min-ed25519.pem"
+var testED25519CAKeyFile = "testdata/5min-ed25519-key.pem"
 
 var invalidCryptoParams = []csr.KeyRequest{
 	// Weak Key
 	{A: "rsa", S: 1024},
 	// Bad param
 	{A: "rsaCrypto", S: 2048},
 	{A: "ecdsa", S: 2000},
+	{A: "ed25519", S: 512},
```

Comment:
The size parameter is ignored for ed25519, so this is not invalid. Let's remove this line.
```suggestion
```

Just to confirm, from your branch it looks like the test is indeed failing:
```
go test ./initca/...
[...]
--- FAIL: TestInvalidCryptoParams (0.00s)
    initca_test.go:240: InitCA with bad params should fail: <nil>
[...]
```

Reply:
While we're at it, could you update the error message on lines 217 and 240? Right now they're checking that the error is `nil` and then printing it, but it would be more useful if they printed the invalid config instead.

Line 217:
```
if err == nil {
-    t.Fatal("InitCA with bad CAConfig should fail:", err)
+    t.Fatalf("InitCA with bad CAConfig should fail: %v", invalidCAConfig)
}
```

Line 240:
```
if err == nil {
-    t.Fatal("InitCA with bad params should fail:", err)
+    t.Fatalf("InitCA with bad params should fail: %v", invalidParam)
}
```

Reply:
updated it.